src/maximum_cardinality_matching.jl

```diff
@@ -0,0 +1,118 @@
+"""
+maximum_cardinality_matching(g::Graph)
+
+Given a graph `g` returns a maximum cardinality matching.
+This is the same as running `maximum_weight_matching` with default weights (`1`)
+but is faster without needing a JuMP solver.
+
+Returns MatchingResult containing:
+  - the maximum cardinality that can be achieved
+  - a list of each vertex's match (or -1 for unmatched vertices)
+"""
+function maximum_cardinality_matching(g::AbstractGraph{U}) where U<:Integer
+    n = nv(g)
+    matching = -ones(Int, n)
+    
+    # the number of edges that can possibly be part of a matching
+    max_generally_possible = fld(n, 2)
+
+    # get initial matching
+    matching_len = 0
+    for e in edges(g)
+        # if unmatched
+        if matching[e.src] == -1 && matching[e.dst] == -1
+            matching[e.src] = e.dst
+            matching[e.dst] = e.src
+            matching_len += 1
+        end
+    end
```

Comment:
Maybe it would be a good idea, to have this as a separate function? Then it could also be used for finding a greed maximal cardinality matching and be used by other functions. We could pass it in as an optional argument or a keyword argument.

Furthermore, I was thinking, maybe it would be more efficient to loop over the vertices and then over all neighbors of that vertex? I.e something like 
```julia
for u in vertices(g)
   matching[u] == -1 || continue
   for v in neighbors[u]
      u <= v && continue
      matching[u] = v
      matching[v] = u
     matching_len += 1
   end
end
```
This approach only works, if we assume that `vertices(g)` is ordered in increasing order, but it can also be adapted, such that the increasing order is not required.

Also, be careful with self-loops, these should never be in a matching.
      

```

Reply:
If neighbors is implemented in a fast manner yes. Thought that looping over the edges might be faster than accessing the neighbors but yours has an earlier break. Maybe I can test it on a larger graph. Good point with self loops!

Reply:
I think for `SimpleGraph`the edges iterator is implemented using `neighbors` anyway.

Reply:
Is now a separate function but still uses `edges` for now.